### PR TITLE
Contexts edge cases5

### PR DESCRIPTION
**Pull request by pmuir**
We need to merge https://github.com/fabric8io/fabric8-ui/pull/267 then I can rebase this one and https://github.com/fabric8io/fabric8-ui/pull/268

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 21ce6a76735a7ce883b082dffb6a6ddac88d4517
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 21ce6a76735a7ce883b082dffb6a6ddac88d4517
  * Your commit message does not contain a body

 * Commit: ba21adf5d96bef976fdd0dae9581016a11870730
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 21ce6a76735a7ce883b082dffb6a6ddac88d4517
  * Your commit message does not contain a body

 * Commit: ba21adf5d96bef976fdd0dae9581016a11870730
  * Your commit message does not contain a body

 * Commit: b0b8d6a6fac1e4c7fa3c55f06faac85bbfbf07cc
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 21ce6a76735a7ce883b082dffb6a6ddac88d4517
  * Your commit message does not contain a body

 * Commit: ba21adf5d96bef976fdd0dae9581016a11870730
  * Your commit message does not contain a body

 * Commit: b0b8d6a6fac1e4c7fa3c55f06faac85bbfbf07cc
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 21ce6a76735a7ce883b082dffb6a6ddac88d4517
  * Your commit message does not contain a body

 * Commit: ba21adf5d96bef976fdd0dae9581016a11870730
  * Your commit message does not contain a body

 * Commit: b0b8d6a6fac1e4c7fa3c55f06faac85bbfbf07cc
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 21ce6a76735a7ce883b082dffb6a6ddac88d4517
  * Your commit message does not contain a body

 * Commit: ba21adf5d96bef976fdd0dae9581016a11870730
  * Your commit message does not contain a body

 * Commit: b0b8d6a6fac1e4c7fa3c55f06faac85bbfbf07cc
  * Your commit message does not contain a body

 * Commit: 5584a176da7ae5fa2226eb14e3da82398262ff30
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: fe21ea2b3257541c32c9c025b085e1203c56e078
  * Your commit message does not contain a body

 * Commit: 7179d12139dbc76955f615c85d852d83beb5cc79
  * Your commit message does not contain a body

 * Commit: 80becef818bb10e3558b2c1415751180e652c65b
  * Your commit message does not contain a body

 * Commit: 7a5a04925d0b39221ffa43d6820289416d86d12f
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

File: src/app/shared/login.service.ts [hunk 1]
```diff
@@ -1,47 +1,89 @@
-import { WIT_API_URL } from 'ngx-fabric8-wit';
-
 import { Router } from '@angular/router';
 import { LocalStorageService } from 'angular-2-local-storage';
 import { Injectable, Inject } from '@angular/core';
-import { Headers, Http } from '@angular/http';
 
-import 'rxjs/add/operator/toPromise';
+import { Observable } from 'rxjs';
```
Comment (joshuawilson) [inline src/app/shared/login.service.ts]: not used

File: src/app/shared/login.service.ts [hunk 2]
```diff
@@ -1,47 +1,89 @@
...
+
+import { Broadcaster, AuthenticationService } from 'ngx-login-client';
+import { WIT_API_URL } from 'ngx-fabric8-wit';
+
+import { ContextService } from './context.service';
+import { Navigation } from './../models/navigation';
```
Comment (joshuawilson) [inline src/app/shared/login.service.ts]: not used

File: src/app/shared/login.service.ts [hunk 3]
```diff
@@ -1,47 +1,89 @@
...
 
   constructor(
-    private http: Http,
     private router: Router,
     private localStorage: LocalStorageService,
-    @Inject(WIT_API_URL) apiUrl: string
+    @Inject(WIT_API_URL) apiUrl: string,
+    private broadcaster: Broadcaster,
+    private authService: AuthenticationService,
+    private contextService: ContextService
```
Comment (joshuawilson) [inline src/app/shared/login.service.ts]: not used

File: src/app/shared/spaces.service.ts [hunk 4]
```diff
@@ -1,19 +1,112 @@
-import { Observable } from 'rxjs';
-import { Spaces, Contexts, Space } from 'ngx-fabric8-wit';
+import { ConnectableObservable } from 'rxjs/observable/ConnectableObservable';
+import { ReplaySubject } from 'rxjs/ReplaySubject';
 import { Injectable } from '@angular/core';
 
+import { LocalStorageService } from 'angular-2-local-storage';
+import { Observable, Subject, BehaviorSubject } from 'rxjs';
```
Comment (joshuawilson) [inline src/app/shared/spaces.service.ts]: BehaviorSubject no used

**Pull request merged**